mail/resourceAvailability.go

```diff
@@ -0,0 +1,117 @@
+package mail
+
+import (
+	"fmt"
+	"github.com/PPSKSY-Cluster/backend/db"
+	"github.com/go-co-op/gocron"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+)
+
+var s = gocron.NewScheduler(time.Local)
+
+var check = func() {
+	if err := checkNotifications(); err != nil {
+		log.Println(err.Error())
+	}
+}
+
+func InitSchedule() error {
+
+	every, err := strconv.Atoi(os.Getenv("NOTIFICATION_INTERVAL"))
+	if err != nil {
+		return err
+	}
+
+	if s.Len() == 0 {
```

Comment:
Well yes there would be no errors for the route because it would be part of the servers startup process to start the schedule and if it fails the server wouldn't start. Which to me doesn't seem so wrong.

The reason it only runs once is that in the line
```
notificationRouter.Post("/", notificationCreateHandler())
```
notificationCreateHandler is actually executed and the return value (the actual handler, which currently executes InitSchedule) is subscribed as the handler to "/" and only that function is called if someone accesses the route. Meaning anything before the return handler is run just once. This can be nice for i.e. creating route specific structs as seen in /api/login I think or in previous versions of the auth package where we read the pem file once on startup

Reply:
Ah I see, Ill implement it this way then